### PR TITLE
refactor: adjust to the design components that use UiList

### LINKED ISSUE #53: [Refactor]: adjust components that use UiList to the design
Components should be in accordance with the design.
- UiAccordion
- UiMenu
- UiMultipleAnswer
- UiMultipleChoices 
- UiStepper

### PR DESCRIPTION
**Pull request by ArkadiuszKrzymienski**
# Related issue

close #53

# Scope of work
Adjust to the design components that use UiList:
- UiAccordion before changes:

- UiAccordion after changes:

- UiMenu before changes:

- UiMenu after changes:

- UiStepper before changes:

- UiStepper after changes: 

- UiMultipleAnswer before changes:

- UiMultipleAnswer after changes:

- UiMultipleChoices before changes:

- UiMultipleChoices after changes:

### CODE REVIEW

File: src/components/organisms/UiList/_internal/UiListItemSuffixAsButton.vue [hunk 1]
```diff
@@ -79,3 +79,18 @@ const defaultProps = computed(() => ({
...
 const hasIcon = computed(() => (!!defaultProps.value.iconSuffixAttrs.icon));
 </script>
+
+<style lang="scss">
+@use "../../../../styles/functions";
+
+.ui-list-item-suffix-as-button {
+  $element: list-item-suffix-as-button;
+
+  gap: functions.var($element, gap, var(--space-4));
```
Comment (pspaczek) [inline src/components/organisms/UiList/_internal/UiListItemSuffixAsButton.vue:89]: @ArkadiuszKrzymienski you have right about this approach, `gap` is awesome 🙈 . When you use `visuall-hidden on label you have always right spacing

File: src/components/organisms/UiList/_internal/UiListItemSuffixAsButton.vue [hunk 2]
```diff
@@ -8,7 +8,10 @@
         labelAttrs
       }"
     >
-      <span v-bind="labelAttrs">{{ label }}</span>
+      <span
+        v-if="label"
```
Comment (pspaczek) [inline src/components/organisms/UiList/_internal/UiListItemSuffixAsButton.vue:12]: we miss that 🙈

File: src/components/organisms/UiMultipleAnswer/UiMultipleAnswer.vue [hunk 3]
```diff
@@ -184,13 +183,13 @@ const hintType = computed<'error'|'default'>(() => (props.touched && props.inval
 watch(valid, (value) => {
   emit('update:invalid', !value);
 }, { immediate: true });
-const updateHandler = (newValue: string | unknown[] | Record<string, unknown>) => {
+const handleValueUpdate = (newValue: string | unknown[] | Record<string, unknown>) => {
```
Comment (pspaczek) [inline src/components/organisms/UiMultipleAnswer/UiMultipleAnswer.vue:186]: I thing about use this name convention https://jaketrent.com/post/naming-event-handlers-react/ WDYT?

File: src/components/organisms/UiMultipleAnswer/_internal/UiMultipleAnswerItem.vue [hunk 4]
```diff
@@ -2,25 +2,25 @@
...
+    :value="value"
+    :model-value="modelValue"
+    :text-label-attrs="defaultProps.textLabelAttrs"
+    :suffix-attrs="defaultProps.suffixAttrs"
     :class="[
       'ui-multiple-answer-item', errorClass
     ]"
-    :value="value"
-    :model-value="modelValue"
-    v-bind="defaultProps"
```
Comment (pspaczek) [inline src/components/organisms/UiMultipleAnswer/_internal/UiMultipleAnswerItem.vue]: IMO 
```
:text-label-attrs="defaultProps.textLabelAttrs"
 :suffix-attrs="defaultProps.suffixAttrs"
```
is more redable than `defaultProps`. You know what you bind. I can find value for use `v-bind="defaultProps"`

File: src/components/organisms/UiMultipleAnswer/_internal/UiMultipleAnswerItem.vue [hunk 5]
```diff
@@ -2,25 +2,25 @@
...
     <template #content>
       <slot
         v-bind="{
           id,
-          modelValue,
           value,
-          updateHandler,
           invalid,
           label,
+          component,
```
Comment (pspaczek) [inline src/components/organisms/UiMultipleAnswer/_internal/UiMultipleAnswerItem.vue:23]: yup is required to use `choice` :D

```
<template #choice="{
      id,
      value,
      invalid,
      label,
      component,
    }">
      <component
        :id="id"
        :is="component"
        v-model="modelValue"
        :value="value"
        :class="[
          'ui-list-item__content',
          { 'ui-checkbox--has-error ui-list-item--has-error': invalid }
        ]"
      >
        {{ label }}
      </component>
    </template>
```

File: src/components/organisms/UiMultipleAnswer/_internal/UiMultipleAnswerItem.vue [hunk 6]
```diff
@@ -114,38 +114,54 @@ const isCheckbox = computed(() => (Array.isArray(props.modelValue)));
...
+  const input: HTMLInputElement | null | undefined = suffixButton.closest('.ui-multiple-answer-item')?.querySelector('input');
+  if (input) {
+    event.preventDefault();
+    focusElement(input);
+  }
 }
 function focusSuffix(event: KeyboardEvent) {
   if (event.key !== 'ArrowRight') return;
   const input = event.target as HTMLInputElement;
-  const suffixBtn: HTMLElement | null | undefined = input.closest('.ui-multiple-answer-item')?.querySelector('.ui-list-item-suffix');
```
Comment (pspaczek) [inline src/components/organisms/UiMultipleAnswer/_internal/UiMultipleAnswerItem.vue]: `.ui-list-item-suffix` 🛑  you should add class to this element :) and use it in this function. When you use `.ui-list-item-suffix` you create dependency and when you change something in `.ui-list-item-suffix` you should change it here :(

File: src/components/organisms/UiMultipleAnswer/_internal/UiMultipleAnswerItem.vue [hunk 7]
```diff
@@ -165,12 +181,10 @@ if (choiceItem.value) {
   $this: &;
   $element: multiple-answer-item;
 
-  .ui-checkbox,
```
Comment (pspaczek) [inline src/components/organisms/UiMultipleAnswer/_internal/UiMultipleAnswerItem.vue]: 🛑  you should add class related to `ui-multiple-answer-item` I add `ui-multiple-answer-item__label` to handle both cases :) and you miss `align-items: flex-start;` ;)

File: src/components/organisms/UiMultipleAnswer/_internal/UiMultipleAnswerItem.vue [hunk 8]
```diff
@@ -101,80 +86,129 @@ const props = defineProps({
...
+   * Use this props to pass attrs for label UiText.
+   */
+  textLabelAttrs: {
+    type: Object as PropType<MultipleAnswerLabelAttrs>,
+    default: () => ({}),
+  },
   /**
    * Use this props to pass attrs for info UiButton.
    */
   buttonInfoAttrs: {
```
Comment (pspaczek) [inline src/components/organisms/UiMultipleAnswer/_internal/UiMultipleAnswerItem.vue:105]: @ArkadiuszKrzymienski after go to review UiMultipleChoices I think we should keep the same API for both elements and
* keep buttonInfoAttrs etc...
* translate buttonInfoAttrs etc... to suffixAttrs
When we use name Button Info in UiMultipleChoices we should use the same name here (the same feature)

File: src/components/molecules/UiStepper/_internal/UiStepperStep.vue [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+
+<style lang="scss">
+@use "../../../../styles/functions";
+@use "../../../../styles/mixins";
+
+.ui-stepper-step {
+  $this: &;
+  $element: stepper-step;
+
+  --_stepper-step-indicator-width: #{functions.var($element + "-step-indicator", width, 4px)};
```
Comment (AlanLes) [inline src/components/molecules/UiStepper/_internal/UiStepperStep.vue:116]: Q: It looks weird, what's this pattern? Why is this named like this `--_...`?

**Review by AlanLes — APPROVED** [review]
Looks nice ;) Just one quetion :) 
BTW. Why do we continuously write stories in `.js` and not `.ts` files?

File: src/components/molecules/UiStepper/_internal/UiStepperStep.vue (hunk 9 above)
Comment (pspaczek) [reply src/components/molecules/UiStepper/_internal/UiStepperStep.vue:116]: private variable to use inside component but not to override by user :)

File: src/components/organisms/UiMultipleChoices/_internal/UiMultipleChoicesItem.vue [hunk 10]
```diff
@@ -308,107 +351,56 @@ const optionsToRender = computed(() => props.options.map((option) => ({
...
-
-      [dir="rtl"] & {
-        margin: functions.var($element + "-rtl-tablet-option", margin, 0 var(--space-24) 0 0);
-
-        &:first-of-type {
-          margin: functions.var($element + "-rtl-tablet-option", margin, 0);
-        }
-      }
+      margin: functions.var($element + "-tablet-header", margin, 0);
+      gap: functions.var($element + "-tablet-header", gap, var(--space-8));
```
Comment (pspaczek) [inline src/components/organisms/UiMultipleChoices/_internal/UiMultipleChoicesItem.vue:395]: @ArkadiuszKrzymienski thanks for showing me how `gap` is useful. I love it now <3

**Review by pspaczek — APPROVED** [review]
:thank_you: @ArkadiuszKrzymienski

File: src/components/molecules/UiStepper/_internal/UiStepperStep.vue (hunk 9 above)
Comment (AlanLes) [reply src/components/molecules/UiStepper/_internal/UiStepperStep.vue:116]: @pspaczek and does it work? I mean, is the user really not able to override it? 🤔 😄

**Pull request merged**